src/clp_ffi_js/ir/LogEventWithLevel.hpp

```diff
@@ -0,0 +1,45 @@
+#ifndef CLP_FFI_JS_IR_LOG_EVENT_WITH_LEVEL_HPP
+#define CLP_FFI_JS_IR_LOG_EVENT_WITH_LEVEL_HPP
+
+#include <utility>
+
+#include <clp/ir/EncodedTextAst.hpp>
+#include <clp/ir/LogEvent.hpp>
+#include <clp/ir/types.hpp>
+#include <clp/time_types.hpp>
+
+#include <clp_ffi_js/constants.hpp>
+
+namespace clp_ffi_js::ir {
+
+/**
+ * A class derived from LogEvent with an additional member for log level.
```

Comment:
```suggestion
 * A class derived from `clp::ir::LogEvent` with an additional member for the log level.
 *
 * NOTE: Once we move to the next IR format, this class should no longer be necessary since each
 * IR log event will contain a set of key-value pairs, one of which should be the log level.
```